### PR TITLE
xpu optimization for generation case

### PR DESCRIPTION
**Pull request by sywangyi**

### CODE REVIEW

**Comment by sywangyi** [comment]
@liangan1 @vasqu @LuFinch pls help to review this pr.

**Comment by LuFinch** [comment]
LGTM.
I have verified that 
1. SDPA can run into `enable_gqa=True` for XPU now.
2. SDPA can run into `is_causal=True` for XPU when using Static KVcache.

**Comment by sywangyi** [comment]
xpu sdpa kernel does not have the limitation as cuda that is_causal should be set False when seqlen q not equal kv

**Review by liangan1 — APPROVED** [review]

File: src/transformers/integrations/sdpa_attention.py [hunk 1]
```diff
@@ -29,6 +30,9 @@ def use_gqa_in_sdpa(attention_mask: Optional[torch.Tensor], key: torch.Tensor) -
     # 1. torch version >= 2.5
     # 2. attention_mask is None (otherwise it will fall back to the math kernel)
     # 3. key is not a torch.fx.Proxy (otherwise it will fail with a tracing error)
+    if is_torch_xpu_available():
+        # torch xpu support gpa since 2.8, and does not attention_mask limitation
```
Comment (vasqu) [inline src/transformers/integrations/sdpa_attention.py]: Nit: Can we move the comments on the conditions together, e.g.
```
GQA can only be used under the following conditions
1. cuda
   - torch version >= 2.5
   - attention_mask is None (otherwise it will fall back to the math kernel)
   - key is not a torch.fx.Proxy (otherwise it will fail with a tracing error)
2. xpu
...
```

File: src/transformers/masking_utils.py [hunk 2]
```diff
@@ -225,11 +226,18 @@ def _ignore_causal_mask_sdpa(
     if (
         not is_tracing
         # only cases when lower and upper diags are the same, see https://github.com/pytorch/pytorch/issues/108108
-        and (query_length == 1 or kv_length == query_length)
+        and (query_length == 1 or (kv_length == query_length if not is_torch_xpu_available() else True))
```
Comment (vasqu) [inline src/transformers/masking_utils.py]:
```suggestion
        and (query_length == 1 or (kv_length == query_length or is_torch_xpu_available()))
```

File: src/transformers/masking_utils.py [hunk 3]
```diff
@@ -225,11 +226,18 @@ def _ignore_causal_mask_sdpa(
...
         # in this case we need to add special patterns to the mask so cannot be skipped otherwise
         and (local_attention_size is None or kv_length < local_attention_size)
         # In this case, we need to add padding to the mask, so cannot be skipped otherwise
-        and (padding_mask is None or padding_mask.all())
+        and (
+            padding_mask is None
+            or (
+                padding_mask.all()
+                if not is_torch_xpu_available() or query_length == 1
+                else padding_mask[:, :query_length].all()
```
Comment (vasqu) [inline src/transformers/masking_utils.py]: Do we need the or condition here? I think it can be left out, no? So `padding_mask.all() if not is_torch_xpu_available() else padding_mask[:, :query_length].all()`

**Review by vasqu — COMMENTED** [review]
Mostly nits and a question on a conditional

**Comment by vasqu** [comment]
Could you also take a look into the CI failures, seem like there are some dynamo issues :eyes:

File: src/transformers/masking_utils.py (hunk 3 above)
Comment (sywangyi) [reply src/transformers/masking_utils.py]: no, for query_length == 1 path, it's for 2nd+ token generation. attention mask should be set correctly. so we need the original logic.
for query_length > 1, path, it's for 1nd tokenizer generation, is_causal could be set, no attention_mask needed if padding_mask[:, :query_length].all() = True.

**Comment by sywangyi** [comment]
> Could you also take a look into the CI failures, seem like there are some dynamo issues 👀


I have fixed the dynamo ci issue

**Review by ArthurZucker — APPROVED** [review]
Thanks! 🤗
